### PR TITLE
chore: More bootstrap config refactoring

### PR DESCRIPTION
**Pull request by nrcventura**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description

Moves the code in the configuration partial class into the bootstrap config (it is only used at bootstrap time and not used after that so dynamic updates do not matter).

This results in the following:
- A more consistent management of the AgentEnabled and AgentEnabledAt settings.
- Elimination of the unused Xml property so that we do not need to keep the original xml string in memory
- Elimination of the LogConfig property, which is replaced with the original `log` field
- Elimination of the configurationPartial partial class

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Core/Config/BootstrapConfiguration.cs [hunk 1]
```diff
@@ -56,5 +134,55 @@ private bool CheckServerlessModeEnabled(configuration localConfiguration)
...
+            {
+                var stringValueWithProvenance = getStringValueWithProvenance(Constants.AppSettingsAgentEnabled);
+                if (stringValueWithProvenance != null && stringValueWithProvenance.Value != null && bool.TryParse(stringValueWithProvenance.Value, out var value))
+                {
+                    return new ValueWithProvenance<bool>(value, stringValueWithProvenance.Provenance);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error($"Failed to read {Constants.AppSettingsAgentEnabled} from local config.", ex);
```
Comment (chynesNR) [inline src/Agent/NewRelic/Agent/Core/Config/BootstrapConfiguration.cs:182]: I think you have this syntax backwards: the exception should be the first parameter.

**Review by chynesNR — APPROVED** [review]
Looks good!

File: src/Agent/NewRelic/Agent/Core/Config/BootstrapConfiguration.cs (hunk 1 above)
Comment (nrcventura) [reply src/Agent/NewRelic/Agent/Core/Config/BootstrapConfiguration.cs:182]: Good catch! I didn't look closely enough at the copilot output.

**Comment by nrcventura** [comment]
The MySql test failure is a known flicker, and will be addressed separately.

**Pull request merged**